### ISSUE TITLE
sidebar can behave erratically due to underlying neovim errors

### ISSUE THREAD

**Issue by Akin909** [onivim/oni#2005]
This issue is primarily for documentation/discussion although if a potential way around this issue is discovered that would be great. 

An issue I noticed with the shared neovim instance behaving strangely accounting for #1396 etc. actually relate to an obscure yet I can imagine definitely possible to crop up issue with starting the sharedNeovimInstance.

In my specific case I have `automizu/LanguageServerClient` which I use/used in terminal vim, this plugin registers an `rplugin` which neovim seems to always check for registered rplugins in `$XDG_HOME/rplugins.vim` on startup even if passed a different file as vimrc to load, the `rplugins.vim` is created when `UpdateRemotePlugins` is run.

In my case the last time I ran this command a manifest was created in the terminal and settings some conditional to not run these plugins in oni still left the hanging references in rplugins.vim, which tbh I think is more of an issue with the plugins see here autozimu/LanguageClient-neovim#198.

TLDR;

I think there can be external elements which can conflict unexpectedly with neovim's startup that might need to be looked into.

I appreciate this is kind of a unique circumstance but the way in which I imagine this recurring is as remote plugins possibly/probably increase in popularity it might be the case that weird issues like this pop up.

Possible consideration should oni startup the shared neovim instance with a `--noplugins` flag, not sure that would work with rplugins but seems like a place to start.

**Comment by justinmk** [comment]
It's a high priority task for Nvim to have better handling of errors so that they don't interfere with external clients/UIs. It's been a pain for a long time and I'm hoping to find time to work on it soon.
